common/sshtools.py

```diff
@@ -412,6 +413,37 @@ def unlockSshAgent(self, force=False):
                     env['ASKPASS_TEMP'] = thread.temp_file
                     thread.start()
 
+                from pprint import pprint
+                with open('env.test', 'w') as file:
+                    pprint(env, stream=file)
+
+                # We need to validate the ssh key password which
+                # `backintime-askpass` will provide before calling
+                # ssh-add below.
+
+                # Validate cached SSH key password:
+                proc = subprocess.run(
+                        ['ssh-keygen', '-y', '-f', self.private_key_file],
+                        capture_output=True,
+                        # Ensure ssh-keygen uses backintime-askpass
+                        env=env | {'SSH_ASKPASS_REQUIRE': 'prefer'}
+                )
+
+                # if backintime-askpass supplied an invalid cached password
+                if proc.returncode > 0:
+                    pw = password.Password()
+
+                    # TODO: Figure out correct way to get pw_id
+                    pw_id = 1
+
```

Comment:
TODO: What does `pw_id` mean? How do we know what the correct value is?

Reply:
I need to do some more digging here. `qt/settingsdialog.py` talks about `password_1` and `password_2`, and this seems directly related to `pw_id = 1` and `pw_id = 2`. I'm wondering if one is the password for the SSH key, and the other is the encryption password?

Reply:
It depends on the profile mode. Using "SSH encrypted" you will get this in the Generals tab.
![image](https://github.com/user-attachments/assets/50887471-cb99-4ffc-be6a-274cc05f6c61)
"SSH private key" is "password_1" and "Encryption" is "password_2". You can see this relation in the beginning of `Config.saveProfile()`.

Reply:
I've added a comment explaining where this comes from in c3bd0a132.